src/main/kotlin/com/chloe/kotlinserv/Batch.kt

```diff
@@ -0,0 +1,33 @@
+package com.chloe.kotlinserv
+
+import com.zaxxer.hikari.HikariDataSource
+
+class Batch(private val ds: HikariDataSource, private val query: String) : Runnable {
```

Comment:
lets don't extend Runnable but do batch executions inside of this class (see comments below)

Also I think it's better to rename it to something like GeoDataWriter (do also interface for it) with methods addGeoData(geoData, ipAddress), flush() - flush will write list to clickhouse and clear it